docs/source/observing.rst

```diff
@@ -0,0 +1,67 @@
+Observing
+=========
+
+This package includes some utilities for observing the progress and performance of running pipelines. In a similar
+manner to the python logging framework, observers are meant to be used as a global facility whose handling can be
+transparently swapped out by the user to send results to different backends.
+
+For example, a task writer can add observers to critical parts of code::
+
+  from dplutils import observer
+
+  def my_task(indf):
+      rows_in = len(indf)
+      with observer.timer('critical_section'):
+          ...
+      rows_out = len(indf)
+      observer.observe('filter_percent', rows_out / rows_in * 100)
+
+
+By default the above would do nothing when that task is executed in a pipeline, since we do not know what the user would
```

Comment:
The statement "By default, the above would do nothing when that task is executed in a pipeline" could perhaps be made clearer. Could it be intended to convey that no metrics are recorded unless an observer is explicitly set?
If so, it might be helpful to add titles to distinguish the purposes of the two sections more clearly:
Example Usage
Setting an Observer